### ISSUE TITLE
Improve site performance

### ISSUE THREAD

**Issue by Roesh** [blitz-js/blitzjs.com#470]
Brandon used lightouse and found that we could improve performance:

https://lighthouse-dot-webdotdevsite.appspot.com//lh/html?url=https%3A%2F%2Fblitzjs.com

[app/pages/index](https://github.com/blitz-js/blitzjs.com/blob/e6f46efd1502e8c98c137c69cf7fac4653466a5d/app/pages/index.js#L227) loads several github avatars using a plain img tag. In order to use an Image component, we need to add the domains we are receiving the images from to module.exports

```
images: {
    domains: [
      "avatars.githubusercontent.com",
      "avatars0.githubusercontent.com",
      "avatars1.githubusercontent.com",
...
```
Doing domains: [ *.githubusercontent.com"], but that doesn't work. I tried to see if github has like a certain number of these like maybe 5 max, but they don't say the number:

> To host your images, GitHub uses the open-source project Camo. Camo generates an anonymous URL proxy for each image which hides your browser details and related information from other users. The URL starts https://<subdomain>.githubusercontent.com/, with different subdomains depending on how you uploaded the image. 

[Info about avatar storage](https://docs.github.com/en/github/authenticating-to-github/about-anonymized-image-urls#troubleshooting-issues-with-camo)

There are a few next steps I see here:
1) Add regex support to the images.domains array so we can load any subdomain easily
2) Reach out to the github team to ask about how many subdomains there could be max, and if their names may change.
3) @MrLeebo suggested replacing `<img>` with `<Image>` will **not** really benefit things for githubusercontent.com. github is almost certainly already optimizing the images and hosting them on CDNs, so our images are probably better off as `<img loading="lazy">` elements
4) @MrLeebo also mentioned that if we could make improvements to github's own hosting, we could do something like 
```<Image src={`/avatar?src=${encodeURIComponent(contributor.avatar_url)}`} />```
and set up a custom route to proxy the githubusercontent domain so you don't have to whitelist each of them

I am going to start with the third option of lazy loading images and seeing if that improves our performance sufficiently, and work on the fourth option if it does not.

5) I reviewed the react-player npm package and have added changes to improve performance. The only drawback is that it takes two clicks to play the video. Initially, there is an image with the video thumbnail. A user's first click loads the actual youtube player, and the second click actually starts the video.

**Comment by Roesh** [comment]
@dawnofmidnight pointed out that the [Sponsor links](https://github.com/blitz-js/blitzjs.com/blob/287b7ba2943f570971a480d3c59b8de3edc65d90/app/core/components/SponsorPack.js) also were not optimized for image loading

```
<div
                            key={`circle-${i}`}
                            className="absolute bg-no-repeat bg-center bg-contain rounded-sm"
                            style={{
                              left: "50%",
                              top: "50%",
                              transform: "translate(-50%, -50%)",
                              width: circle.data.cost > 100 ? "80%" : "50%",
                              height: circle.data.cost > 100 ? "80%" : "50%",
                              backgroundImage: `url(${circle.data.imageUrl})`, <---------------
                            }}
                          />
```

**Comment by MrLeebo** [comment]
@Roesh You mischaracterized what I said on Discord. I was taking the opposite position; I don't think converting avatar.githubusercontent.com content to `<Image />` would be beneficial. I suspect that github already has a process to host, transform, and cache those avatar pictures at scale. If I'm wrong, you could proxy the requests so that you wouldn't have to whitelist their CDNs, but I would be very surprised if `<Image />` is doing something that Github doesn't already do.

**Comment by JuanM04** [comment]
@MrLeebo `<Image />` serves .webp and lower resolutions... right?

**Comment by Roesh** [comment]
@MrLeebo sorry, I made a typo. Fixed the comment.

@JuanM04 I wasn't really able to test the performance improvement of the change I made, so I'll push just the one with the img tags but loaded lazily, and see how the lighthouse report changes.

I also wanted to replace the background urls one with `<Image />` but I don't know enough CSS to do that - the images are squashed and misshaped. If someone else would like to take a stab at that it would be good
